src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -71,23 +71,29 @@ internal static (string? Prefix, string Alias) SplitPrefix(this string rawAlias)
         internal static TokenizeResult Tokenize(
             this IReadOnlyList<string> args,
             CommandLineConfiguration configuration,
-            bool inferRootCommand = true)
+            bool inferRootCommand)
         {
-            var errorList = new List<string>();
+            const int FirstArgIsNotRootCommand = -1;
+
+            List<string>? errorList = null;
```

Comment:
usually there are no errors, so we delay the allocation until first error is found